sdk/extensions/Azure.Extensions.AspNetCore.Configuration.Secrets/tests/AzureKeyVaultConfigurationTests.cs

```diff
@@ -700,5 +723,20 @@ private class JsonKeyVaultSecretManager: KeyVaultSecretManager
                 return data;
             }
         }
+
+        private class SingleJsonSecretKeyVaultSecretManager : JsonKeyVaultSecretManager
+        {
+            private readonly string _name;
+
+            public SingleJsonSecretKeyVaultSecretManager(string name)
+            {
+                _name = name;
+            }
+
+            public override AsyncPageable<SecretProperties> LoadSecretPropertiesAsync(SecretClient client, CancellationToken cancellationToken = default)
+            {
+                return client.GetPropertiesOfSecretVersionsAsync(_name, cancellationToken);
```

Comment:
Why are you querying for all versions of a secret here when you only need the latest (since no version is provided to the class)? This seems wasteful, since not only is the payload larger but multiple fetches from the service may be necessary. If the goal was to minimize network-bound IO, just return a single secret's properties e.g.

```c#
KeyVaultSecret secret = await client.GetSecretAsync(_name, cancellationToken: cancellationToken);
yield return secret.Properties;
```